core/controllers/resources_test.py

```diff
@@ -272,6 +273,76 @@ def test_audio_upload(self):
         )
         self.logout()
 
+    def test_audio_upload_with_non_mp3_file(self):
+        self.login(self.EDITOR_EMAIL)
+        response = self.get_html_response('/create/0')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with open(
+            os.path.join(feconf.TESTS_DATA_DIR, self.TEST_AUDIO_FILE_FLAC_1),
+            mode='rb') as f:
+            raw_audio = f.read()
+
+        self.post_json(
+            '%s/0' % (self.AUDIO_UPLOAD_URL_PREFIX),
+            {'filename': self.TEST_AUDIO_FILE_FLAC_1},
+            csrf_token=csrf_token,
+            upload_files=(('raw_audio_file', 'unused_filename', raw_audio),)
```

Comment:
Let's use a list-of-tuples here instead of a tuple-of-tuples? It's easier to extend and harder to get wrong (since the last comma isn't required).

Reply:
Done